hepcrawl/spiders/s3_springer_spider.py

```diff
@@ -60,61 +61,86 @@ class S3SpringerSpider(XMLFeedSpider):
+        self.ftp_port = ftp_port
+
+        # Creating target folders
+        paths_of_folders = [
+            os.path.join(SPRINGER_DOWNLOAD_DIR, 'EPJC'),
+            os.path.join(SPRINGER_DOWNLOAD_DIR, 'JHEP'), 
+            os.path.join(SPRINGER_UNPACK_FOLDER, 'EPJC'), 
+            os.path.join(SPRINGER_UNPACK_FOLDER, 'JHEP')
+            ]
+
+        for path_of_folder in paths_of_folders: 
+            if not os.path.exists(path_of_folder):
+                os.makedirs(path_of_folder)
+
     def start_requests(self):
         """List selected folder on remote FTP and yield new zip files."""
-        if self.package_path:
-            ftp_params = {"ftp_local_filename": self.package_path}
-            yield Request(self.package_path, meta=ftp_params, callback=self.handle_package_ftp)
-        else:
-            ftp_host, ftp_params = ftp_connection_info(self.ftp_host, self.ftp_netrc)
-            missing_files = []
-            for journal in ['EPJC', 'JHEP']:
-                _, tmp_missing_files = sftp_list_files(
-                    os.path.join(self.ftp_folder, journal),
-                    os.path.join(self.target_folder, journal),
-                    server=ftp_host,
-                    user=ftp_params['ftp_user'],
-                    password=ftp_params['ftp_password']
-                )
-                missing_files.extend(tmp_missing_files)
-            # TODO - add checking if the package was already downloaded
-
-            for remote_file in missing_files:
-                journal = 'EPJC' if 'EPJC' in remote_file else 'JHEP'
-                remote_file = str(remote_file).strip('/data/in/%s/' % journal)
-                ftp_params["ftp_local_filename"] = os.path.join(
-                    self.target_folder,
-                    journal,
-                    remote_file
-                )
-                remote_url = "ftp://{0}/{1}".format(ftp_host, os.path.join('data/in/', journal, remote_file))
-                yield Request(
-                    str(remote_url),
-                    meta=ftp_params,
-                    callback=self.handle_package_ftp)
 
-    def handle_package_ftp(self, response):
+        self.log('Harvest started.', logging.INFO)
+        if self.package_path:
+            self.log('Harvesting locally: %s' %
+                     self.package_path, logging.INFO)
+            return [Request(self.package_path, callback=self.handle_package_sftp), ]
+        return self.download_files_from_sftp()
+
+    def download_files_from_sftp(self):
+        sftp_host, sftp_params = ftp_connection_info(
+            self.ftp_host, self.ftp_netrc)
+        cnopts = pysftp.CnOpts()
+        cnopts.hostkeys = None
+
+        self.log("Connecting to SFTP server...", logging.INFO)
+        # Connect to the ftp server
+        with pysftp.Connection(sftp_host, username=sftp_params['ftp_user'], password=sftp_params['ftp_password'],
+                               port=self.ftp_port, cnopts=cnopts) as sftp:
```

Comment:
we don't need to pass the `port` it's `22` by default https://pysftp.readthedocs.io/en/release_0.2.8/pysftp.html, we should remove it also from ``__init__.py``